app/assets/javascripts/thyme/thyme_player.js

```diff
@@ -0,0 +1,212 @@
+  annotationsToggle.add();
+  (new EmergencyButton('emergency-button')).add();
+  (new FullScreenButton('full-screen', thymeContainer)).add();
+  (new MinusButton('minus-ten', 10)).add();
+  (new MuteButton('mute')).add();
+  (new NextChapterButton('next-chapter')).add();
+  (new PlayButton('play-pause')).add();
+  (new PlusButton('plus-ten', 10)).add();
+  (new PreviousChapterButton('previous-chapter')).add();
+  (new SpeedSelector('speed')).add();
+  // initialize iaButton here to have the reference but call add() later
+  // when we can define toHide (second argument which is set to null here)
+  const iaButton = new IaButton('ia-active', null, [$(video), $('#video-controlBar')], '82%');
+  (new VolumeBar('volume-bar')).add();
+  seekBar = new SeekBar('seek-bar');
+  seekBar.add();
+  seekBar.addChapterTooltips();
+
+
+
+  /*
+    ANNOTATION FUNCTIONALITY
+   */
+  // annotation manager and area
+  function colorFunc(annotation) {
+    return annotation.color;
+  }
+  function isValid(annotation) {
+    return (!annotationsToggle.getValue() && !annotation.belongsToCurrentUser ? false : true);
+  }
+  const annotationArea = new AnnotationArea(true, colorFunc, isValid);
+  thymeAttributes.annotationArea = annotationArea;
+  function strokeColorFunc(annotation) {
+    return 'black';
+  }
+  function sizeFunc(annotation) {
+    return false;
+  }
+  function onClick(annotation) {
+    annotationArea.update(annotation);
+    annotationArea.show();
+    $('#caption').hide();
+  }
+  function onUpdate() {
+    /* update might change the annotation which is currently shown in the
+       annotation area -> find the updated annotation in the annotation array
+       and update the area. */
+    if (annotationArea.annotation != null) {
+      const id = annotationArea.annotation.id;
+      annotationArea.update(AnnotationManager.find(id));
+    }
+  }
+  const annotationManager = new AnnotationManager(colorFunc, strokeColorFunc, sizeFunc,
+                                                  onClick, onUpdate, isValid);
+  thymeAttributes.annotationManager = annotationManager;
+  // onShow and onUpdate definition for the annotation area
+  function onShow() {
+    iaButton.plus();
+    annotationManager.updateMarkers();
+  }
+  function onHide() {
+    iaButton.minus();
+    annotationManager.updateMarkers();
+    resizeContainer();
+  }
+  annotationArea.onShow = onShow;
+  annotationArea.onHide = onHide;
+
+  // Update annotations after submitting the annotations form
+  $(document).on('click', '#annotation-modal-submit-button', function() {
+    /* NOTE:
+       Updating might take some time on the backend,
+       so I added a slight delay.
+       I couldn't think of an easy way to let the script
+       wait for the update to complete (as with the delete button),
+       but it might be possible! */
+    setTimeout(function() {
+      annotationManager.updateAnnotations();
+    }, 500);
+  });
```

Comment:
I don't fully understand, how you would do it. I need something that triggers the updateAnnotations() method, once the backend has updated the corresponding annotation. How can I send a success response from the update method of the AnnotationsController to thyme_player.js?

Reply:
But this is a JS file from the view. My action has to contact a JS file from the asset folder.

A hidden HTML element would be a possible (although maybe not the best) solution for that problem, because it can serve as a bridge between view JS and asset JS.

Reply:
I updated it now in the way, I tried to describe above.

Reply:
Your change did not work for me in a chromium-based browser since the event you were using is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/MutationEvent). I've rewritten the logic as proposed without any additional HTML elements, see commit 8fcc2389.